spec/controllers/concerns/sign_in_duration_concern_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe SignInDurationConcern, type: :controller do
+  let(:test_class) do
+    Class.new do
+      include SignInDurationConcern
+
+      attr_reader :session
+
+      def initialize(session = {})
+        @session = session
+      end
+    end
+  end
+
+  let(:instance) { test_class.new }
+
+  describe '#sign_in_duration_seconds' do
+    before do
+      instance.session[:sign_in_page_visited_at] = 6.seconds.ago.to_s
+    end
+
+    it 'returns 6 seconds' do
+      expect(instance.sign_in_duration_seconds).to eq(6)
+    end
+
+    it 'returns nil' do
+      instance.session[:sign_in_page_visited_at] = nil
+      expect(instance.sign_in_duration_seconds).to eq(nil)
+    end
```

Comment:
1. I think we'll want to `freeze_time` again here to avoid flakey spec failures due to time drift
2. I think we should use `context` to describe when we're expecting "6 seconds" vs. expecting "nil", right not it's unclear
3. `6.seconds.ago.to_s` is a string, which is different from what we're actually assigning to the session in `SessionsController` (which is a date). We should make sure the value is accurate to what's expected to be assigned in actual usage

```suggestion
    let(:sign_in_page_visited_at) {}

    around do |example|
      freeze_time { example.run }
    end

    before do
      instance.session[:sign_in_page_visited_at] = sign_in_page_visited_at
    end

    context 'when session value is assigned' do
      let(:sign_in_page_visited_at) { 6.seconds.ago }

      it 'returns seconds since value' do
        expect(instance.sign_in_duration_seconds).to eq(6)
      end
    end

    context 'when session value is not assigned' do
      let(:sign_in_page_visited_at) { nil }

      it 'returns nil' do
        expect(instance.sign_in_duration_seconds).to be_nil
      end
    end
```